src/include/buffer.h

```diff
@@ -440,7 +440,13 @@ struct error_code;
 	buffers_iterator(U* const p)
 	  : cur(p) {
 	}
-	template <class U>
+	// copy constructor
+	buffers_iterator(const buffers_iterator<T>& other)
+	  : cur(other.cur) {
+	}
+	// converting constructor, from iterator -> const_iterator only
+	template <class U, std::enable_if_t<
```

Comment:
i have following failure:
```
2022-03-17T05:44:34.277 INFO:tasks.workunit.client.0.smithi067.stderr:In file included from /usr/include/rados/librados.hpp:11,
2022-03-17T05:44:34.277 INFO:tasks.workunit.client.0.smithi067.stderr:                 from /home/ubuntu/cephtest/mnt.0/client.0/tmp/rocksdb/include/rocksdb/utilities/env_librados.h:14,
2022-03-17T05:44:34.278 INFO:tasks.workunit.client.0.smithi067.stderr:                 from /home/ubuntu/cephtest/mnt.0/client.0/tmp/rocksdb/utilities/env_librados.cc:4:
2022-03-17T05:44:34.278 INFO:tasks.workunit.client.0.smithi067.stderr:/usr/include/rados/buffer.h:448:26: error: ‘std::enable_if_t’ has not been declared
2022-03-17T05:44:34.278 INFO:tasks.workunit.client.0.smithi067.stderr:  template <class U, std::enable_if_t<
2022-03-17T05:44:34.278 INFO:tasks.workunit.client.0.smithi067.stderr:                          ^~~~~~~~~~~
2022-03-17T05:44:34.279 INFO:tasks.workunit.client.0.smithi067.stderr:/usr/include/rados/buffer.h:448:37: error: expected ‘>’ before ‘<’ token
2022-03-17T05:44:34.279 INFO:tasks.workunit.client.0.smithi067.stderr:  template <class U, std::enable_if_t<
2022-03-17T05:44:34.279 INFO:tasks.workunit.client.0.smithi067.stderr:                                     ^
2022-03-17T05:44:34.282 INFO:tasks.workunit.client.0.smithi067.stderr:/usr/include/rados/buffer.h: In member function ‘void ceph::buffer::v15_2_0::list::buffers_t::insert_after(ceph::buffer::v15_2_0::list::buffers_t::const_iterator, ceph::buffer::v15_2_0::list::buffers_t::reference)’:
```

see https://pulpito.ceph.com/kchai-2022-03-17_05:18:57-rados-wip-cxx20-fixes-core-kefu-distro-default-smithi/6740698/

Reply:
as rocksdb is still a C++11 project. we have to stick with `std::enable_if<>`.

Reply:
is it possible to enable c++20 just for `env_librados.cc`?

Reply:
on second thought, i agree that it's better to support c++11 librados clients

Reply:
indeed, rocksdb is just one of them.

Reply:
@cbodley i am going to squash the fix-up change into your commit if it's fine by you.

Reply:
i was wrong. turns out rocksdb switched to C++17 in https://github.com/facebook/rocksdb/commit/fd3e0f43b3cd041c38e37543e0b616d18ca55179.